### PR TITLE
Add Datastore resource-prefix header

### PR DESCRIPTION
**Pull request by blowmage**
Add `google-cloud-resource-prefix` header to GRPC requests.

- [x] Datastore
- [x] Language
- [x] Logging
- [x] Pub/Sub
- [x] Speech
- [x] Vision

### CODE REVIEW

**Comment by blowmage** [comment]
@pcostell Can you verify this is the change you are looking for? Thanks!

**Comment by quartzmo** [comment]
Just a note:

The decision to use GAX `CallOptions#kwargs` is directed by @jmuk in [this comment](https://github.com/GoogleCloudPlatform/google-cloud-ruby/issues/869#issuecomment-245815685).

File: google-cloud-datastore/lib/google/cloud/datastore/service.rb [hunk 1]
```diff
@@ -144,6 +153,14 @@ def generate_read_options consistency, transaction
           nil
         end
 
+        def default_headers
+          { "google-cloud-resource-prefix" => "projects/#{@project}" }
```
Comment (pcostell) [inline google-cloud-datastore/lib/google/cloud/datastore/service.rb]: Sorry for my lack of ruby-foo -- just to double check that for project "my-project" this turns into: "projects/my-project"?

Comment (quartzmo) [inline google-cloud-datastore/lib/google/cloud/datastore/service.rb]: Yes, correct.

Comment (blowmage) [inline google-cloud-datastore/lib/google/cloud/datastore/service.rb]: Yep. That is correct. `#{}` is the shortcut for string interpolation/

**Comment by pcostell** [comment]
LGTM

**Comment by quartzmo** [comment]
I'll merge when green. (C'mon Travis CI)

**Comment by blowmage** [comment]
K, all GRPC services have the resource-prefix header. This is good to merge whenever the builds decide to finish.

**Pull request merged**